src/mcpbr/config.py

```
@@ -71,6 +110,129 @@ def get_expanded_env(self) -> dict[str, str]:
+
+    @field_validator("resource_group")
+    @classmethod
+    def validate_resource_group(cls, v: str) -> str:
+        """Validate resource group name format.
+
+        Azure resource groups must contain only alphanumeric characters, dashes, and underscores.
+        """
+        if not re.match(r"^[a-zA-Z0-9_-]+$", v):
+            raise ValueError(
+                "resource_group must contain only alphanumeric characters, dashes, and underscores"
+            )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)